ray-curator/tests/stages/deduplication/conftest.py

```diff
@@ -0,0 +1,120 @@
+"""Shared test configuration for Ray Curator tests.
+
+This module provides shared Ray cluster setup and teardown for all test modules.
+Using a single Ray instance across tests improves performance while maintaining
+proper isolation through Ray's actor/task lifecycle management.
+TODO: Merge this with ray-curator/tests/backends/conftest.py
+"""
+
+import os
+import subprocess
+
+import pytest
+import ray
+from loguru import logger
+
+
+def find_free_port():
+    import socket
+
+    with socket.socket(socket.AF_INET, socket.SOCK_STREAM) as s:
+        s.bind(("", 0))
+        return s.getsockname()[1]
+
+
+@pytest.fixture(scope="session", autouse=True)
+def shared_ray_cluster(tmp_path_factory: pytest.TempPathFactory):
+    """Set up a shared Ray cluster for all tests in the session.
+
+    This fixture automatically sets up Ray at the beginning of the test session
+    and tears it down at the end. It configures Ray with fixed resources for
+    consistent testing behavior.
+    """
+
+    ONE_GB = 1024**3  # noqa: N806
+
+    # This ensures we are not reusing an existing cluster but starting a new one
+    if "RAY_ADDRESS" in os.environ:
+        del os.environ["RAY_ADDRESS"]
+
+    # Create a temporary directory for Ray to avoid conflicts with other instances
+    temp_dir = tmp_path_factory.mktemp("ray1")
+
+    # TODO: Create a cluster with 11 cpus instead of one head node with 11 cpus
+    # If we have 6-7 stages all needing 1 cpu, then we atleast need 10 cpus for Xenna / Ray Data to work
+    # The 11th CPU is for StageCallCounter
+
+    ray_port = find_free_port()
+    dashboard_port = find_free_port()
+    ray_client_server_port = find_free_port()
+
+    # TODO: See if we can use get_client in the future
+    cmd_to_run = [
+        "ray",
+        "start",
+        "--head",
+        "--disable-usage-stats",
+        "--port",
+        str(ray_port),
+        "--dashboard-port",
+        str(dashboard_port),
+        "--ray-client-server-port",
+        str(ray_client_server_port),
+        "--temp-dir",
+        str(temp_dir),
+        "--num-cpus",
+        "11",
+        "--num-gpus",
+        "2",
```

Comment:
Should not hard code here, either  set it to min of current gpu available. We can get number of gpus using something like below. 


```suggestion
        def get_gpu_count():
            pynvml.nvmlInit()
            return pynvml.nvmlDeviceGetCount()
```

Reply:
Will keep it as is for now. Not using dynamic allocation for both cpu and gpu because otherwise on bigger clusters it'll just end up using all the CPUs / GPUs for unittests (like our lab machines which isn't ideal)